lib/transactionTracking/evm/useEVMTracking.ts

```diff
@@ -0,0 +1,34 @@
+import { useWaitForTransaction } from "wagmi"
+import { NetworkType } from "../../../Models/CryptoNetwork"
+import { useSettingsState } from "../../../context/settings"
+import { TransactionStatus, TxTrackingProvider } from "../../../hooks/useTransactionTracking"
+import { Layer } from "../../../Models/Layer"
+
+export default function useEVMTracking(hash: `0x${string}`, network: Layer): TxTrackingProvider {
+
+    const { layers } = useSettingsState()
+    const supportedNetworks = [
+        ...layers.filter(layer => layer.type === NetworkType.EVM).map(l => l.internal_name),
+    ]
+
+    const transaction = useWaitForTransaction({
```

Comment:
- All providers will trigger wait for transaction need to check if the provider supports the network or not